clarity-ext-scripts/clarity_ext_scripts/covid/partner_api_client.py

```diff
@@ -255,12 +266,101 @@ def search_for_service_request(self, org, org_referral_code):
                     ("More than one partner referral code was found for organization: {} "
                      "and organization referral code: {}").format(org, org_referral_code))
             else:
+                log.debug("Response json was: {}".format(response_json))
                 raise OrganizationReferralCodeNotFound(
                     ("No partner referral code was found for organization: {} "
                      "and organization referral code: {}").format(org, org_referral_code))
         except PartnerClientAPIException as e:
             log.error(e.message)
             raise e
 
-    def post_diagnosis_report(self, diagnosis_report):
-        raise NotImplementedError
+    def post_diagnosis_report(self, service_request_id, diagnosis_result, analysis_results):
+        try:
+            payload = self._create_payload(
+                service_request_id, diagnosis_result, analysis_results)
+            diagnosis_result_url = "{}/DiagnosticReport".format(
+                self._base_url)
+            log.debug("URL: {}".format(diagnosis_result_url))
```

Comment:
This read a bit to me like that I should expect two urls, another one called analysis_results_url (since this url maps with the variable name `diagnosis_result`. Maybe just call it `url` instead?